src/box/lua/session.c

```diff
@@ -203,6 +205,8 @@ lbox_session_su(struct lua_State *L)
 	struct credentials su_credentials;
 	struct credentials *old_credentials = fiber()->storage.credentials;
 	credentials_create(&su_credentials, user);
+	uint32_t auth_uid = session->credentials.auth_uid;
+	su_credentials.auth_uid = auth_uid;
```

Comment:
AFAIU you could distinguish `effective` and `auth` credentials by looking at `fiber()->storage.credentials` and `fiber()->storage.session.credentials` respectively.

Could you make it so in case of `box.session.su(user)` as well? Now `box.session.su(user)` strangely rewrites session credentials and puts them to fiber credentials. If this wasn't the case, you wouldn't need the new `auth_uid` field at all. You would simply look at `storage.credentials` for euid and `session.credentials` for uid, or auth_uid, whatever you call it.